src/terms.jl

```diff
@@ -468,20 +496,27 @@ julia> modelcols(MatrixTerm(ts), d)
 """
 modelcols(ts::TupleTerm, d::NamedTuple) = modelcols.(ts, Ref(d))
 
-# TODO: @generated to unroll the getfield stuff
-modelcols(ft::FunctionTerm{Fo,Fa,Names}, d::NamedTuple) where {Fo,Fa,Names} =
-    ft.fanon.(getfield.(Ref(d), Names)...)
 
-modelcols(t::ContinuousTerm, d::NamedTuple) = Float64.(d[t.sym])
+# TODO: @generated to unroll the accessing of fields stuff
+function modelcols(ft::FunctionCallTerm, d::ColumnTable)
+    cols = (modelcols(term, d) for term in ft.terms)
+    return ft.funct.(cols...)
+end
 
+modelcols(t::ContinuousTerm, d::NamedTuple) = Float64.(d[t.sym])
 modelcols(t::CategoricalTerm, d::NamedTuple) = t.contrasts[d[t.sym], :]
 
+# This can exist in a formula after apply_schema if it was protected
+modelcols(t::ConstantTerm, d::ColumnTable) = fill(t.n, length(d))
+modelcols(t::ConstantTerm, d::Tables.RowTable) = fill(t.n, length(first(d)))
```

Comment:
I think rowtable and column table are switched actually.
(that would explain my dimension mismatch errors)

A RowTable isn't a Row though. that would be length 1.
Our RowTable handling is AFAICT kinda not real, but this isn't the place for that discussion

Reply:
Right.  Just do the right thing for column table and leave row table handling for the future.  And add a method for `NamedTuple` that just returns the scalar so that it works for a single row?